analytics/aggregate.go

```diff
@@ -179,10 +182,60 @@ func (a *SQLAnalyticsRecordAggregate) ProcessStatusCodes() {
 	a.Counter.ErrorMap = nil
 }
 
-func (ar *SQLAnalyticsRecordAggregate) TableName() string {
+func (f *SQLAnalyticsRecordAggregate) TableName() string {
 	return AggregateSQLTable
 }
 
+func (f SQLAnalyticsRecordAggregate) GetAssignments(tableName string) map[string]interface{} {
+	assignments := make(map[string]interface{})
+
+	baseFields := structs.Fields(f)
+	for _, field := range baseFields {
+		colName := field.Tag("json")
+		if strings.Contains(colName, "code_") {
+			if !field.IsZero() {
+				assignments[colName] = gorm.Expr(tableName + "." + colName + " + " + fmt.Sprint(field.Value()))
+			}
+		}
+	}
+
+	fields := structs.Fields(f.Counter)
+	for _, field := range fields {
+		jsonTag := field.Tag("json")
+		colName := "counter_" + jsonTag
+
+		switch jsonTag {
+		//hits, error, success"s, open_connections, closed_connections, bytes_in, bytes_out,total_request_time, total_upstream_latency
+		case "hits", "error", "success", "open_connections", "closed_connections", "bytes_in", "bytes_out", "total_request_time", "total_upstream_latency":
+			if !field.IsZero() {
+				assignments[colName] = gorm.Expr(tableName + "." + colName + " + " + fmt.Sprintf("%v", field.Value()))
+			}
+		//request_time, upstream_latency
+		case "request_time", "upstream_latency", "latency":
+			//AVG adding value to another AVG: newAve = ((oldAve*oldNumPoints) + x)/(oldNumPoints+1)
+			if !field.IsZero() {
+				assignments[colName] = gorm.Expr("((" + tableName + "." + colName + " * " + tableName + ".counter_hits) + " + fmt.Sprintf("%v", field.Value()) + ")/( " + tableName + ".counter_hits +1)")
```

Comment:
Frankly I thought this logic will be done on Dashboard side. 
How it is implemented for mongo, we also calculate it on mongo side?

Reply:
@buger we do it on the pump side, this would be a combination of `AsChange` and `AsTimeUpdate`.
The difference is in mongo agg pump we get the doc back and recalculate the AVG for `"request_time", "upstream_latency", "latency"` in an easier way ( TotalVal / hits). In SQL, since we don't know the value that's in the table we need to do it with that math.